### PR TITLE
chore: client migration

### PR DESCRIPTION
**Pull request by SameedUllah**
Migration Completed from JS to TS

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -10,8 +10,10 @@
     "@types/node": "^16.11.27",
     "@types/react": "^18.0.6",
     "@types/react-dom": "^18.0.2",
+    "nodemon": "^2.0.15",
```
Comment (SafiUllahAshfaq) [inline client/package.json]: `nodemon` should be a `devDependency`.

File: client/src/components/AddNote.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const AddNote = () => {
+
+
```
Comment (SafiUllahAshfaq) [inline client/src/components/AddNote.tsx]: Are you sure this file is prettified?

File: client/src/components/NotesList.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import AddNote from './AddNote';
+import Note from './Note';
+import Search from './Search';
+
+
+const notesList = () => {
+	return (
+		<div className='notes-list'>
```
Comment (SafiUllahAshfaq) [inline client/src/components/NotesList.tsx]: Is this file prettified?

File: client/src/components/NotesList.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+            <Note/>
+            <Note/>
+            <Note/>
+            <AddNote/>
+
+		</div>
+	);
+};
+
+export default notesList;
```
Comment (SafiUllahAshfaq) [inline client/src/components/NotesList.tsx]: No new line at the end of file.

**Review by SafiUllahAshfaq — CHANGES_REQUESTED** [review]

**Comment by SafiUllahAshfaq** [comment]
Can you please install [Semantic Pull Requests](https://probot.github.io/apps/semantic-pull-requests/) extension on this repo?

**Review by SafiUllahAshfaq — COMMENTED** [review]
Can you please share the reason behind deleting most of the files?

**Comment by SameedUllah** [comment]
> Can you please share the reason behind deleting most of the files?

I thought that they were irrelevant but not sure.

**Comment by SafiUllahAshfaq** [comment]
If they were created by `create-react-app`, then leave them as is.

File: client/package.json [hunk 5]
```diff
@@ -39,5 +40,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "nodemon": "^2.0.15"
```
Comment (SafiUllahAshfaq) [inline client/package.json]: is `nodemon` being used anywhere in the project?

File: client/public/index.html [hunk 6]
```diff
@@ -1,43 +1,16 @@
...
 
-      To begin the development, run `npm start` or `yarn start`.
-      To create a production bundle, use `npm run build` or `yarn build`.
-    -->
-  </body>
-</html>
+  </div>
+</body>
+
+</html>
```
Comment (SafiUllahAshfaq) [inline client/public/index.html]: No new line at end of file.

File: client/src/components/Note.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { MdDeleteForever } from "react-icons/md";
+
+const Note = () => {
+  var date = new Date().getDate();
```
Comment (SafiUllahAshfaq) [inline client/src/components/Note.tsx]:
```suggestion
  var date = new Date();
```

`new Date().getDate()` will return day of the month only.

**Review by SafiUllahAshfaq — CHANGES_REQUESTED** [review]

File: client/src/components/NotesList.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+const notesList = () => {
+  return (
+    <div className="notes-list">
+      {/* <Search /> */}
+      <AddNote />
+      <Note />
+      <Note />
+      <Note />
+      <Note />
+      <Note />
```
Comment (SafiUllahAshfaq) [inline client/src/components/NotesList.tsx]: try to use `notes.map(....)` for rendering multiple notes.

**Review by SafiUllahAshfaq — CHANGES_REQUESTED** [review]

File: client/public/index.html (hunk 6 above)
Comment (SameedUllah) [reply client/public/index.html]: Added a new line at the end of this file.

File: client/src/components/NotesList.tsx (hunk 8 above)
Comment (SameedUllah) [reply client/src/components/NotesList.tsx]: Okay

**Review by SafiUllahAshfaq — APPROVED** [review]

**Pull request merged**